test/integration/cds_integration_test.cc

```diff
@@ -0,0 +1,211 @@
+#include "envoy/api/v2/cds.pb.h"
+#include "envoy/api/v2/discovery.pb.h"
+#include "envoy/grpc/status.h"
+#include "envoy/stats/scope.h"
+
+#include "common/config/protobuf_link_hacks.h"
+#include "common/config/resources.h"
+#include "common/protobuf/protobuf.h"
+#include "common/protobuf/utility.h"
+
+#include "test/common/grpc/grpc_client_integration.h"
+#include "test/integration/http_integration.h"
+#include "test/integration/utility.h"
+#include "test/mocks/server/mocks.h"
+#include "test/test_common/network_utility.h"
+#include "test/test_common/simulated_time_system.h"
+#include "test/test_common/utility.h"
+
+#include "gtest/gtest.h"
+
+using testing::AssertionFailure;
+using testing::AssertionResult;
+using testing::AssertionSuccess;
+using testing::IsSubstring;
+
+namespace Envoy {
+namespace {
+
+const std::string kConfig = R"EOF(
+admin:
+  access_log_path: /dev/null
+  address:
+    socket_address:
+      address: 127.0.0.1
```

Comment:
I would think so, but I got a "JSON at lines 1-1 does not conform to schema" when I tried. Maybe there's some validation code somewhere that didn't get updated to permit a missing address field? (I can go hunt that down, but I figured I'd ask you first in case it's immediately obvious to you what/where is wrong.)

Reply:
I think if you completely elide the admin proto you should be fine. I know for sure that Envoy can load with an empty config file now.

Reply:
I have now become aware of the HTTP_PROXY_CONFIG default bootstrap config over in test/config/utility.cc. Working from the ADS test, which specifies its own config, I didn't realize that existed! Envoy itself might be able to start without the admin proto, but I don't think the integration test framework is: 26/29 of the integration tests failed when I removed the admin proto. Maybe I should keep it for now in this new test, and then a separate PR can break the dependency and remove it?

Reply:
OK, it might be needed for `/quitquitquit` shutdown actually. Just seems like weird boilerplate; maybe we should define a `EMPTY_PROXY_CONFIG` string somewhere for this and use it here?